src/queries/text.js

```diff
@@ -9,6 +9,31 @@ import {
   buildQueries,
 } from './all-utils'
 
+function ignoreHiddenElements(container, selector, baseArray) {
+  const hiddenElements = [
+    ...Array.from(container.querySelectorAll("[aria-hidden='true']")),
+  ]
+  baseArray.forEach(base => {
+    if (base.getAttribute('aria-hidden') === 'true') {
+      hiddenElements.push(base)
+    }
+  })
+  let foundHiddenChildren = [...hiddenElements]
+  hiddenElements.forEach(hiddenElement => {
+    foundHiddenChildren = [
+      ...foundHiddenChildren,
+      ...Array.from(hiddenElement.querySelectorAll(selector)),
+    ]
+  })
+  const findSameNode = foundElement =>
+    !foundHiddenChildren.find(foundHiddenElement =>
+      foundElement.isSameNode(foundHiddenElement),
+    )
```

Comment:
`.find` returns the first element from an array that matches the provided callback. In this case you don't particularly care what the element is, just whether it can be found or not. That behaviour is better provided by `.some` which just returns a boolean saying whether the given callback could be matched anywhere in the array:
```suggestion
  const findSameNode = foundElement =>
    !foundHiddenChildren.some(foundHiddenElement =>
      foundElement.isSameNode(foundHiddenElement),
    )
```

Additionally the naming of `findSameNode` doesn't quite feel right to me... I think I would call this something more like "isElementAccessible"